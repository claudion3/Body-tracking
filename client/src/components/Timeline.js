import React from 'react';
import './Timeline.css';
import {
	VerticalTimeline,
	VerticalTimelineElement,
} from 'react-vertical-timeline-component';
import 'react-vertical-timeline-component/style.min.css';
import { Link } from 'react-scroll';

const Timeline = ({ data, removeTruck, editTruck }) => {
	const date = new Date(data.date).toLocaleDateString();

	return (
		<VerticalTimeline data-testid='timeline'>
			<VerticalTimelineElement
				contentStyle={{ background: ' #E9E6EA', color: '#000' }}
				contentArrowStyle={{ borderRight: '7px solid  rgb(233,230,234)' }}
				date={date}
				iconStyle={{ background: 'rgb(16, 204, 82)', color: '#fff' }}>
				<h3 className='vertical-timeline-element-title'>Body Truck Data</h3>

				<p className='vertical-timeline-element-subtitle'>
					Weight {data.weight} Kg
				</p>
				<p className='vertical-timeline-element-subtitle'>
					Hip Width {data.hipWidth} cm
				</p>
				<div className='edit-delete'>
					<Link to='edit' smooth={true} duration={1000}>
						<button className='edit' onClick={() => editTruck(data)}>
							<i class='fas fa-edit'></i>
						</button>
					</Link>

					<button className='delete' onClick={() => removeTruck(data._id)}>
						<i className='fas fa-trash'></i>
					</button>
				</div>
			</VerticalTimelineElement>
		</VerticalTimeline>
	);
};

export default Timeline;
