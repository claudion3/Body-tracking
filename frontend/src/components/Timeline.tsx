// import React from 'react';

// interface TimelineEntry {
//   date: string;
//   weight: number;
//   hipSize: number;
// }

// interface TimelineProps {
//   entries: TimelineEntry[];
// }

// const Timeline: React.FC<TimelineProps> = ({ entries }) => {
//   if (entries.length === 0) {
//     return (
//       <div className="text-center text-gray-600">
//         No progress entries found. Start tracking your progress!
//       </div>
//     );
//   }

//   return (
//     <div className="space-y-4">
//       {entries.map((entry, index) => (
//         <div key={index} className="flex items-start space-x-4">
//           <div className="flex-shrink-0">
//             <div className="w-4 h-4 bg-blue-500 rounded-full"></div>
//           </div>
//           <div className="flex-1">
//             <div className="text-lg font-semibold">{entry.date}</div>
//             <div className="text-sm text-gray-600">Weight: {entry.weight} kg</div>
//             <div className="text-sm text-gray-600">Hip Size: {entry.hipSize} cm</div>
//           </div>
//         </div>
//       ))}
//     </div>
//   );
// };

// export default Timeline;