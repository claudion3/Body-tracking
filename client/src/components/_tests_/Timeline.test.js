import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Timeline from '../Timeline';

afterEach(() => {
	cleanup();
});

const dataTest = [
	{
		_id: '6149c00c8bb94a3b61e23012',
		date: '2021-09-21T00:00:00.000Z',
		weight: 85,
	},
	{
		_id: '6149c01a8bb94a3b61e23015',
		date: '2021-09-23T00:00:00.000Z',
		weight: 70,
	},
];

describe('Navbar component', () => {
	it('Matches snapshot', () => {
		const tree = renderer.create(<Timeline data={dataTest} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
