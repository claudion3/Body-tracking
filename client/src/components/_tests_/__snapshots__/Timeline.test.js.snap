// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Matches snapshot 1`] = `
<div
  className="vertical-timeline vertical-timeline--animate vertical-timeline--two-columns"
>
  <div
    className="vertical-timeline-element"
    id=""
    style={null}
  >
    <span
      className="vertical-timeline-element-icon is-hidden"
      onClick={null}
      style={
        Object {
          "background": "rgb(16, 204, 82)",
          "color": "#fff",
        }
      }
    />
    <div
      className="vertical-timeline-element-content is-hidden"
      onClick={null}
      style={
        Object {
          "background": " #E9E6EA",
          "color": "#000",
        }
      }
    >
      <div
        className="vertical-timeline-element-content-arrow"
        style={
          Object {
            "borderRight": "7px solid  rgb(233,230,234)",
          }
        }
      />
      <h3
        className="vertical-timeline-element-title"
      >
        Body Tracking Data
      </h3>
      <p
        className="vertical-timeline-element-subtitle"
      >
        Weight 
         Kg
      </p>
      <p
        className="vertical-timeline-element-subtitle"
      >
        Hip Width 
         cm
      </p>
      <div
        className="edit-delete"
      >
        <a
          onClick={[Function]}
        >
          <button
            className="edit"
            onClick={[Function]}
          >
            <i
              class="fas fa-edit"
            />
          </button>
        </a>
        <button
          className="delete"
          onClick={[Function]}
        >
          <i
            className="fas fa-trash"
          />
        </button>
      </div>
      <span
        className="vertical-timeline-element-date"
      >
        Invalid Date
      </span>
    </div>
  </div>
</div>
`;
